module/motion/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -264,50 +269,69 @@ namespace module::motion {
-        std::tie(trunk_pos, trunk_axis, foot_pos, foot_axis, is_left_support) = walk_engine.computeCartesianPosition();
+        std::tie(rTSs, Thetast, rFSs, Thetasf, is_left_support) = walk_engine.compute_cartesian_position();
 
         // Change goals from support foot based coordinate system to trunk based coordinate system
-        Eigen::Affine3f Hst;  // trunk_to_support_foot_goal
-        Hst.linear()      = setRPY(trunk_axis.x(), trunk_axis.y(), trunk_axis.z()).transpose();
-        Hst.translation() = -Hst.rotation() * trunk_pos;
+        // Trunk {t} from support foot {s}
+        Eigen::Affine3f Hst;
+        Hst.linear()      = rpy2rot<float>(Thetast);
+        Hst.translation() = rTSs;
 
-        Eigen::Affine3f Hfs;  // support_to_flying_foot
-        Hfs.linear()      = setRPY(foot_axis.x(), foot_axis.y(), foot_axis.z());
-        Hfs.translation() = foot_pos;
+        // Flying foot {f} from support foot {s}
+        Eigen::Affine3f Hsf;
+        Hsf.linear()      = rpy2rot<float>(Thetasf);
+        Hsf.translation() = rFSs;
 
-        const Eigen::Affine3f Hft = Hfs * Hst;  // trunk_to_flying_foot_goal
+        // Support foot {s} from trunk {t}
+        const Eigen::Affine3f Hts = Hst.inverse();
 
-        // Calculate leg joints
-        const Eigen::Matrix4f left_foot  = walk_engine.getFootstep().isLeftSupport() ? Hst.matrix() : Hft.matrix();
-        const Eigen::Matrix4f right_foot = walk_engine.getFootstep().isLeftSupport() ? Hft.matrix() : Hst.matrix();
+        // Flying foot {f} from trunk {t}
+        const Eigen::Affine3f Htf = Hts * Hsf;
 
-        const auto joints =
-            calculateLegJoints<float>(kinematicsModel, Eigen::Affine3f(left_foot), Eigen::Affine3f(right_foot));
+        // Get desired transform for left foot {l}
+        const Eigen::Matrix4f Htl = walk_engine.get_footstep().is_left_support() ? Hts.matrix() : Htf.matrix();
 
-        auto waypoints = motion(joints);
+        // Get desired transform for right foot {r}
+        const Eigen::Matrix4f Htr = walk_engine.get_footstep().is_left_support() ? Htf.matrix() : Hts.matrix();
+
+        // Compute inverse kinematics for left and right foot
+        const auto joints = calculateLegJoints<float>(kinematicsModel, Eigen::Affine3f(Htl), Eigen::Affine3f(Htr));
+        auto waypoints    = motion(joints);
         emit(std::move(waypoints));
+
+        // Plot graphs of desired trajectories
+        if (log_level <= NUClear::DEBUG) {
+            emit(graph("Left foot desired position (x,y,z)", Htl(0, 3), Htl(1, 3), Htl(2, 3)));
+            emit(graph("Right foot desired position (x,y,z)", Htr(0, 3), Htr(1, 3), Htr(2, 3)));
+            emit(graph("Left foot desired orientation (r,p,y)",
+                       rot2rpy<float>(Htl.block(0, 0, 3, 3)).x(),
+                       rot2rpy<float>(Htl.block(0, 0, 3, 3)).y(),
+                       rot2rpy<float>(Htl.block(0, 0, 3, 3)).z()));
```

Comment:
Hard to see on my phone but this might look nicer if you call the function beforehand, store it in a variable with ideally a small name, then use it here. Same with one underneath.